client/src/modules/journal/modals/search.modal.js

```diff
@@ -52,42 +57,62 @@ function JournalSearchModalController(Instance, Projects, Notify,
     vm.defaultQueries.account_id = filters.account_id;
   }
 
-  Account.read()
-    .then(function (accounts) {
-      vm.hrAccounts = accounts.reduce(function (aggregate, account) {
-        aggregate[account.id] = String(account.number).concat(' - ', account.label);
-        return aggregate;
-      }, {});
-    })
-    .catch(Notify.handleError);
-
-  Projects.read()
-    .then(function (projects) {
-      vm.projects = projects;
-    })
-    .catch(Notify.handleError);
-
-  // format voucher types and bind to the view
-  Vouchers.transactionType()
-    .then(function (list) {
-      vm.types = list.data.map(function (item) {
-        item.hrText = $translate.instant(item.text);
-        return item;
+  // load all Transaction types
+  TransactionTypes.read()
+    .then(function (types) {
+      types.forEach(function (item) {
+        item.typeText = $translate.instant(item.text);
       });
+      vm.transactionTypes = types;
     })
     .catch(Notify.handleError);
 
   // handle component selection states
   // custom filter account_id - assign the value to the searchQueries object
   vm.onSelectAccount = function onSelectAccount(account) {
     vm.searchQueries.account_id = account.id;
+    displayValues.account_id = String(account.number).concat(' - ', account.label);
   };
 
+  // Set displayLabel if the filters account is defined
+  if (filters.account_id) {
```

Comment:
I still don't understand why this is even necessary.  Can'y we just do:
```js
displayValues.account_id = account.hrLabel;
```
and that be it?

All of these `if` statements are really ugly.  Let me fork this PR and see if I can get a cleaner version.